docker/config_template.yaml

```diff
@@ -29,6 +29,7 @@ services:
     rpc:
       port: 7933
       bindOnLocalHost: ${BIND_ON_LOCALHOST}
+      bindOnIP: ${BIND_ON_IP}
```

Comment:
I believe this docker startup script also needs to be modified to support BIND_ON_IP correctly
https://github.com/uber/cadence/blob/master/docker/start.sh#L56
It defaults to the address returned by `hostname --ip-address` when BIND_ON_LOCALHOST is false - which may be wrong when BIND_ON_IP is explicitly set.

Reply:
When `BIND_ON_IP` is set (assuming != 127.0.0.1) - `hostname --ip-address` might be a more correct approach, you can bind on `0.0.0.0` but connecting to 0.0.0.0 is not very defined...
I propose changing docker `start.sh`:
```
    export HOST_IP=`hostname --ip-address`

    if [ "$BIND_ON_LOCALHOST" == true ] || [ "$BIND_ON_IP" == "127.0.0.1" ]; then
            export BIND_ON_IP="127.0.0.1"
            export HOST_IP="127.0.0.1"
    fi

    # deprecating this env key
    export BIND_ON_LOCALHOST=false
```

Essentially I deprecate `BIND_ON_LOCALHOST` and replace it with BIND_ON_IP, leaving HOST_IP which is used by cassandra and ringpop

Reply:
makes sense to me.